projects/packages/sync/src/modules/class-module.php

```diff
@@ -745,12 +745,13 @@ public function filter_objects_and_metadata_by_size( $type, $objects, $metadata,
 
 			foreach ( $metadata as $key => $metadata_item ) {
 				if ( (int) $metadata_item->{$type . '_id'} === $object_id ) {
-					$metadata_item_size = strlen( maybe_serialize( $metadata_item->meta_value ) );
+					$metadata_item_size = strlen( maybe_serialize( $metadata_item ) );
```

Comment:
I'd suggest adding a stricter check here to ensure the result after `maybe_serialize` is actually a string